### PR TITLE
Add test case for redirect link on "Block and file"

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar Please review

**Comment by TimothyAsirJeyasing** [comment]
Screenshot from 2023-02-20 13-01-08

File: cypress/support.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ Cypress.Commands.add('install', () => {
     if (code !== 0) {
       cy.clickNavLink(['Operators', 'Installed Operators']);
       cy.byLegacyTestID('item-filter').type('Openshift Data Foundation');
-      cy.byTestRows('resource-row').get('td').first().click();
```
Comment (GowthamShanmugam) [inline cypress/support.ts:35]: byTestRows is used in lot of placed, simply removing will break testcases on other places.

Comment (TimothyAsirJeyasing) [reply cypress/support.ts:35]: Currently install is not working for all the places due to usage of resource-row and get td in it.  So there is a new byTestOperatorRow('OpenShift Data Foundation') available which works. Will once again test other test cases also regarding install.

Comment (GowthamShanmugam) [reply cypress/support.ts:35]: byTestRows is used in different places for a different purposes

File: cypress/support.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ Cypress.Commands.add('install', () => {
...
+      cy.byTestOperatorRow('OpenShift Data Foundation').click();
```
Comment (SanjalKatiyar) [inline cypress/support.ts:35]: this seems wrong... not sure why this change is relevant...

File: cypress/support.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ Cypress.Commands.add('install', () => {
       cy.log('Check if storage system was created');
       cy.clickNavLink(['Operators', 'Installed Operators']);
       cy.byLegacyTestID('item-filter').type('Openshift Data Foundation');
-      cy.byTestRows('resource-row').get('td').first().click();
+      cy.byTestOperatorRow('OpenShift Data Foundation').click();
```
Comment (SanjalKatiyar) [inline cypress/support.ts:67]: don;t just update something which is shared by all the tests... this is not needed...

File: cypress/tests/block-pool.spec.ts [hunk 4]
```diff
@@ -32,6 +32,13 @@ describe('Test block pool under ODF UI', () => {
     navigateToBlockPool();
   });
 
+  it('Test redirect link to operator details page', () => {
+    cy.log('Click service link in block pool');
+    cy.byTestID('horizontal-link-Overview').click();
+    cy.byTestID('ocs-link').click();
+    cy.byLegacyTestID('odf-console').contains('odf-console');
+  });
```
Comment (SanjalKatiyar) [inline cypress/tests/block-pool.spec.ts:40]: why are u updating test for block pool ?
as mentioned in the issue, this is for "Block and file" dashboard...

Comment (TimothyAsirJeyasing) [reply cypress/tests/block-pool.spec.ts:40]: There is no specific test case for block and file. So i have added here, I will do it separately .

Comment (SanjalKatiyar) [reply cypress/tests/block-pool.spec.ts:40]: check again... there is:)

Comment (SanjalKatiyar) [reply cypress/tests/block-pool.spec.ts:40]: check again... there is...

File: cypress/support.ts (hunk 3 above)
Comment (TimothyAsirJeyasing) [reply cypress/support.ts:67]: Sure, will update properly

File: cypress/tests/block-pool.spec.ts (hunk 4 above)
Comment (TimothyAsirJeyasing) [reply cypress/tests/block-pool.spec.ts:40]: Sure will do it in the right file

**Comment by TimothyAsirJeyasing** [comment]
Screenshot from 2023-02-23 19-11-44

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar Could you please review

File: cypress/tests/persistent-dashboard-check-spec.ts [hunk 5]
```diff
@@ -89,4 +89,12 @@ describe('Check Persistent Dashboard', () => {
...
     });
   });
+
+  it('Test redirect link to operator details page', () => {
+    cy.byTestID('ocs-link').click();
+    cy.byLegacyTestID('details-actions')
+      .parent()
+      .contains('OpenShift Data Foundation');
+    cy.byLegacyTestID('odf-console').contains('odf-console');
+  });
```
Comment (SanjalKatiyar) [inline cypress/tests/persistent-dashboard-check-spec.ts:99]: add this in the "it" block for details card that we already have...

Comment (TimothyAsirJeyasing) [reply cypress/tests/persistent-dashboard-check-spec.ts]: Done

**Comment by TimothyAsirJeyasing** [comment]
Screenshot from 2023-02-27 20-44-32

File: cypress/tests/persistent-dashboard-check-spec.ts [hunk 6]
```diff
@@ -41,9 +41,19 @@ describe('Check Persistent Dashboard', () => {
       .contains('ocs-storagecluster')
       .scrollIntoView()
       .should('be.visible');
+    cy.log('Check redirect link goes to operator details page');
+    cy.byTestID('ocs-link').click();
+    cy.byLegacyTestID('details-actions')
+      .parent()
+      .contains('OpenShift Data Foundation');
```
Comment (SanjalKatiyar) [inline cypress/tests/persistent-dashboard-check-spec.ts:48]: use `cy.url`

Comment (TimothyAsirJeyasing) [reply cypress/tests/persistent-dashboard-check-spec.ts:48]: Shall i replace with
cy.url().should('include', 'k8s/ns/openshift-storage/operators.coreos.com~v1alpha1~ClusterServiceVersion/odf-operator');

**Comment by TimothyAsirJeyasing** [comment]
ci/prow/odf-console-e2e-aws

**Comment by TimothyAsirJeyasing** [comment]
ci/prow/odf-console-e2e-aws

**Comment by TimothyAsirJeyasing** [comment]
ci/prow/odf-console-e2e-aws

**Comment by TimothyAsirJeyasing** [comment]
Rebased with latest master

**Comment by TimothyAsirJeyasing** [comment]
Re-base is done

**Review by bipuladh — APPROVED** [review]
